pkg/controllers/management/drivers/kontainerdriver/kontainerdriver.go

```diff
@@ -61,47 +63,23 @@ func (l *Lifecycle) Create(obj *v3.KontainerDriver) (runtime.Object, error) {
 		v3.KontainerDriverConditionInactive.True(obj)
 		return obj, nil
 	}
-
-	v3.KontainerDriverConditionDownloaded.Unknown(obj)
-	v3.KontainerDriverConditionInstalled.Unknown(obj)
-
-	// Update status
-	obj, err := l.kontainerDrivers.Update(obj)
-	if err != nil {
-		return nil, err
-	}
-
-	if !obj.Spec.BuiltIn {
-		obj, err = l.download(obj)
-		if err != nil {
-			return nil, err
-		}
-	} else {
+	if obj.Spec.BuiltIn {
 		v3.KontainerDriverConditionDownloaded.True(obj)
 		v3.KontainerDriverConditionInstalled.True(obj)
+	} else {
+		v3.KontainerDriverConditionDownloaded.Unknown(obj)
+		v3.KontainerDriverConditionInstalled.Unknown(obj)
 	}
 
-	// Update status
-	obj, err = l.kontainerDrivers.Update(obj)
-	if err != nil {
-		return nil, err
-	}
-
-	if hasStaticSchema(obj) {
-		return obj, nil
-	}
-
-	err = l.createDynamicSchema(obj)
-	if err != nil {
-		return nil, err
-	}
-
-	v3.KontainerDriverConditionActive.True(obj)
-
 	return obj, nil
 }
 
 func (l *Lifecycle) driverExists(obj *v3.KontainerDriver) bool {
+	if obj.Status.ExecutablePath != "" {
+		if _, err := os.Stat(obj.Status.ExecutablePath); err == nil {
```

Comment:
I don't understand why we are stat'ing the executable here when`.Exists()` already does the same logic.  Can you please explain?

Reply:
As our external kontainer-engine-driver binary is named as `kontainer-engine-driver-huawei` but our driver crd's name is `huaweicontainercloudengine`. In the original `driverExists` function, it will check `kontainer-engine-driver-huaweicontainercloudengine` binary exists because we add prefix for it in https://github.com/rancher/rancher/blob/master/pkg/controllers/management/drivers/kontainer_driver.go#L30

Reply:
So in this case, driver exists will always return false. I would like to check the binary if executable path is not empty.

Reply:
Ok, I've reviewed the code and tested this out.  I would prefer if this was refactored instead to pass the `obj.Status.ExecutablePath` into the `NewKontainerDriver` func so we have a consistent way of stat'ing the binary, rather than manually stat'ing the binary.

Reply:
After a second thought, I would prefer to change all the external driver crd name from `aliyunkubernetescontainerservice` to `aliyun` so we can keep the logic consistent for both machine driver and kontainer driver.

Reply:
nvm, already update the logic for checking driver existence. Using the `obj.status.DisplayName` instead of `obj.Name`.